Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -2552,10 +2552,7 @@ def visit_ModuleNode(self, node):
         return node
 
     def visit_DefNode(self, node):
-        pxd_def = self.scope.lookup(node.name)
-        if pxd_def and (not pxd_def.scope or not pxd_def.scope.is_builtin_scope):
-            pass
-        elif (self.scope.is_module_scope and self.directives['auto_cpdef']
+        if (self.scope.is_module_scope and self.directives['auto_cpdef']
```

Comment:
Why is this a "simplification"? It seems to remove conditions. Are they irrelevant now?

Reply:
The if statement ` if pxd_def and (not pxd_def.scope or not pxd_def.scope.is_builtin_scope):` is executed in `AlignFunctionDefinitions` transform before `AutoCpdefFunctionDefinition` transform. If this part is evaluated there as True in the `AlignFunctionDefinitions` transform, function definition is transformed to cfunction and hence in that case the `visit_DefNode()` is not called anymore. So I considered this part as not relevant any more.